Create ROS2 release repositories/teams for eigenpy, hpp-fcl, pinocchio, TSID

wxmerkt opened the pull request:
Creating ROS2 release repositories for EigenPy, HPP-FCL, Pinocchio, and TSID to release for Foxy, Galactic, Rolling, and Humble. Existing releases (ROS1, legacy-ROS2) will remain with the existing release repositories

cc: @jcarpent / @nim65s for team membership review. I've also added package maintainers that likely won't be running releases (but will get invited to the release repository teams nonetheless)

wxmerkt committed:
Add eigenpy.tf

wxmerkt committed:
Add hpp-fcl.tf

wxmerkt committed:
Add pinocchio.tf

wxmerkt committed:
Add tsid.tf

nuclearsandwich commented on hpp-fcl.tf:
```diff
@@ -0,0 +1,19 @@
+    "nim65s",
+  ]
+  hpp_fcl_repositories = [
+    "hpp-fcl-release",
```
```suggestion
    "hpp_fcl-release",
```
I'll suggest that we use `_` in the team, repository, and terraform file names to match the ROS package naming conventions.

wxmerkt replied:
For ROS1 and thus the package name, there have been exceptions for dashes for packages that are third-party/CMake packages that do not generate ROS messages (i.e. explicitly hpp-fcl and dynamic-graph). Their `package.xml` for instance uses a dash rather than an underscore as a result. There was some previous discussion [here](https://github.com/ros/rosdistro/pull/25488#discussion_r461023884).

I am happy to modify it for an underscore for the team and repository. Would it still be okay to have the package be released with a dash as that is its name? (Hence we wouldn't have to have two package.xml variants either)

nuclearsandwich replied:
> I am happy to modify it for an underscore for the team and repository. Would it still be okay to have the package be released with a dash as that is its name? (Hence we wouldn't have to have two package.xml variants either)

I will use the opportunity to supply just a bit of friction. If we don't take this opportunity to change to a REP-144 compliant package name when would we? Once all ROS 1 distributions are in end-of-support? Never? There is an ever-present Sword of Damocles for a package in ROS which doesn't adhere to the naming conventions as future tools are not guaranteed nor required to support them. Juggling two package.xml files to maintain the status quo where it has already taken hold will expire in 2025 but having a package name that's out-of-spec will last until a hard break is made to correct it. As I interpret REP-144 (and perhaps the REP could do with a clarifying edit if there are other interpretations favored) the name used in package.xml files for third-party packages _must_ also comply with the package naming requirements. what the package calls itself internally or names its headers is 

Since the package name isn't directly part of this PR and you're alright with the elements which are. I'll proceed with making those changes and merging. I am not putting my foot down about the package name but I am hesitant to allow legacy exceptions to apply indefinitely especially when ROS 2 is such a golden opportunity to correct some of them.

wxmerkt replied:
I fully understand - the decision here for either course of action isn't mine to take. For reducing complexity, it would be an easier route vs vendoring as with `yaml_cpp_vendor` or patching the release with extra CMake config files (*). I'll also cc @nim65s in whose purview a different naming would fall.

(*) I've tried a `condition` on the `name` tag to switch the package name between ROS1/ROS2 but that doesn't work since at least catkin-tools checks that there is only one `name` tag in the `package.xml` - hence we'd need to patch it on release and provide extra CMake configs.

nim65s replied:
Sorry guys, COVID is currently preventing me from thinking straight. I'll trust your decisions here.

nuclearsandwich replied:
> I've tried a condition on the name tag to switch the package name between ROS1/ROS2 but that doesn't work since at least catkin-tools checks that there is only one name tag in the package.xml - hence we'd need to patch it on release and provide extra CMake configs.

I believe condition attributes are only supported / standardized on the various dependency tags and on the export/build_type tag. I don't think there's a single-source solution to using different package names between ROS 1 and ROS 2.

nuclearsandwich replied:
> Sorry guys, COVID is currently preventing me from thinking straight. I'll trust your decisions here.

That's awful to hear. I wish you rest and a swift recovery 💌

nuclearsandwich commented on hpp-fcl.tf:
```diff
@@ -0,0 +1,19 @@
+  
+module "hpp_fcl_team" {
+  source       = "./modules/release_team"
+  team_name    = "hpp-fcl"
```
```suggestion
  team_name    = "hpp_fcl"
```

I'll suggest that we use `_` in the team, repository, and terraform file names to match the ROS package naming conventions.

nuclearsandwich reviewed (changes requested):
Since ROS package names use underscores rather than dashes in rosdistro I think it makes sense to normalize the separator to use `_` for the release repository, release team, and terraform file names.

nuclearsandwich committed:
Use underscore in hpp_fcl team and release repo name.

For consistency with the ROS package naming conventions let's use the
underscore as the separator in names.

nuclearsandwich committed:
List usernames in lexicographical order.

nuclearsandwich committed:
Add new release teams and repositories to the lists.

nuclearsandwich reviewed (approved):
With the agreed upon changes and some additional edits, this is good to go!

wxmerkt commented:
Thank you for the swift review, fixes, and merge :-)
